torchtune/models/clip/_position_embeddings.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import math
+
+import torch
+from torch import nn
+
+# TODO (Felipe): add load hooks + interpolation on positional encodings,
+# so max_num_tiles can be variable and a trained model can be adapted to a
+# new value.
+
+
+class TokenPositionalEmbedding(nn.Module):
+    """
+    Token positional embedding for images. Different for every token (patch) in an image.
```

Comment:
As you mentioned offline, I find the use of token and patch interchangeably a bit confusing. I think you can explicitly call out that a patch and token is the same thing and so num of patches ```(tile_size // patch_size)``` is the same as num of tokens. The other reason why this is important is because shapes in the comments use ```num_tokens``` not ```num_patches```

Reply:
The reson I dont call them the same at all times is because we add a CLS token, i.e. num_tokens = num_patches + 1.

Also, a patch is a square, but a token is a flattened tensor after a convolution. In our case, this convolution is just reshaping the image, but in other cases, it may be something like a resnet. So token and patches are not really the same.

I can add more comments to the docstring clarifying it, and avoid using them interchangeably.